### PR TITLE
Elasticsearch 8 compat

### PR DESCRIPTION
**Pull request by nsantacruz**
Branch that enables compatibility with Elasticsearch 8 while still maintaining compatibility with Elasticsearch 6.

### CODE REVIEW

File: helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            - name: VARNISH_HOST
+              value: "varnish-{{ .Values.deployEnv }}-{{ .Release.Revision }}"
+            - name: SLACK_URL
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "sefaria.secrets.slackWebhook" . }}
+                  key: slack-webhook
+            envFrom:
+            - secretRef:
+                name: {{ template "sefaria.secrets.elasticAdmin" . }}
```
Comment (BrendanGalloway) [inline helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml:48]: if this isn't needed in ES6, can remove ref?

File: helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                memory: 9Gi
+              requests:
+                memory: 7Gi
+            env:
+            - name: REDIS_HOST
+              value: "redis-{{ .Values.deployEnv }}"
+            - name: NODEJS_HOST
+              value: "node-{{ .Values.deployEnv }}-{{ .Release.Revision }}"
+            - name: VARNISH_HOST
+              value: "varnish-{{ .Values.deployEnv }}-{{ .Release.Revision }}"
```
Comment (BrendanGalloway) [inline helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml:40]: Addd SEARCH_HOST override to point to ES6 server

File: helm-chart/sefaria-project/templates/rollout/nginx.yaml [hunk 3]
```diff
@@ -96,11 +94,6 @@ spec:
             value: "varnish-{{ .Values.deployEnv }}-{{ .Release.Revision }}"
           - name: SEARCH_HOST
             value: "{{ .Values.nginx.SEARCH_HOST }}"
```
Comment (BrendanGalloway) [inline helm-chart/sefaria-project/templates/rollout/nginx.yaml:96]: needs a envFrom with the elastic user and password

File: helm-chart/sefaria-project/values.yaml [hunk 4]
```diff
@@ -329,9 +329,7 @@ secrets:
     #   SEFARIA_DB:
     #   SEFARIA_DB_USER:
     #   SEFARIA_DB_PASSWORD:
-    #   SEARCH_ADMIN_USER:
-    #   SEARCH_ADMIN_PW:
-    #   SEARCH_ADMIN_K8S:
+    #   SEARCH_URL
```
Comment (BrendanGalloway) [inline helm-chart/sefaria-project/values.yaml:332]: ES6_SEARCH_HOST: 
ES8_SEARCH_HOST:

File: helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
+{{- if .Values.cronJobs.reindexElasticSearch.enabled }}
```
Comment (BrendanGalloway) [inline helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml:1]: maybe set the to cronjobs.reindexElasticSearch6 and  cronjobs.reindexElasticSearch8

File: helm-chart/sefaria-project/values.yaml (hunk 4 above)
Comment (BrendanGalloway) [reply helm-chart/sefaria-project/values.yaml:332]: Maybe add in cronjob.reindexelasticsearch.x

File: helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml (hunk 2 above)
Comment (BrendanGalloway) [reply helm-chart/sefaria-project/templates/cronjob/reindex-elasticsearch-es6.yaml:40]: name: SEARCH_HOST
values: {{ .Values.cronjobs.reindexelasticsearch.ES6 }}

**Review by BrendanGalloway — CHANGES_REQUESTED** [review]

**Review by BrendanGalloway — DISMISSED** [review]

**Review by edamboritz — APPROVED** [review]

**Pull request merged**